plugin/src/main/java/io/github/rysefoxx/pagination/InventoryContents.java

```diff
@@ -2614,12 +2758,29 @@ public void updateDisplayNameForAll(@Nonnegative int slot,
      *
      * @param slot            The slot
      * @param intelligentItem The new IntelligentItem what should be displayed.
-     * @return true if the ItemStack was updated, false if not.
+     * @return true if the IntelligentItem was updated, false if not.
      * @throws IllegalArgumentException if slot greater than 53 or slot greater than inventory size
      */
     public boolean update(@Nonnegative int slot,
                           @NotNull IntelligentItem intelligentItem) throws IllegalArgumentException {
-        return update(slot, intelligentItem.getItemStack());
+
```

Comment:
Now why exactly did you remove my call to the method, and insert the code inside the method?

Reply:
Because when you do it the way you were doing, it would not update the consumer if there was a new consumer instead of only getting just the itemstack